File: lib/medium/remove_array_of_nodes_from_linked_list.dart
import 'package:dark_coding_challenges/easy/list_node.dart';

ListNode<int>? removeArrayOfNodesFromLinkedList(List<int> nums, ListNode<int>? head) {
  final unique = nums.toSet();
  ListNode<int>? dummy = ListNode(0, head);
  ListNode<int>? current = dummy;
  while (current != null && current.next != null) {
    if (unique.contains(current.next?.val)) {
      current.next = current.next?.next;
    } else {
      current = current.next;
    }
  }
  return dummy.next;
}

File: src/medium/RemoveArrayOfNodesFromLinkedList.swift
import easy

func removeArrayOfNodesFromLinkedList(_ nums: [Int], _ head: easy.ListNode<Int>?) -> easy.ListNode<
  Int
>? {
  let unique = Set(nums)
  let dummy: easy.ListNode<Int> = easy.ListNode<Int>(0)
  dummy.next = head
  var current = dummy
  while let node = current.next {
    if let val = node.val, unique.contains(val) {
      current.next = node.next
    } else {
      current = node
    }
  }
  return dummy.next
}
